crates/rnote-cli/src/export.rs

```diff
@@ -0,0 +1,724 @@
+use std::{
+    fmt::Display,
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+use anyhow::Context;
+use clap::{Args, Subcommand, ValueEnum};
+use parry2d_f64::{bounding_volume::Aabb, na::Vector2};
+use rnote_compose::SplitOrder;
+use rnote_engine::{
+    engine::{
+        export::{
+            DocExportFormat, DocExportPrefs, DocPagesExportFormat, DocPagesExportPrefs,
+            SelectionExportFormat, SelectionExportPrefs,
+        },
+        EngineSnapshot,
+    },
+    RnoteEngine,
+};
+use smol::{
+    fs::File,
+    io::{AsyncReadExt, AsyncWriteExt},
+};
+
+use crate::cli::OnConflict;
+use crate::validators;
+
+#[derive(Subcommand, Debug)]
+pub(crate) enum ExportCommands {
+    /// Export the entire document.{n}
+    /// When using --output-file, only one input file can be given.{n}
+    /// The export format is recognized from the file extension of the output file.{n}
+    /// When using --output-format, the file name and path of the rnote file is used with the extension changed.{n}
+    /// --output-file and --output-format are mutually exclusive but one of them is required.{n}
+    /// Currently `.svg`, `.xopp` and `.pdf` are supported.{n}
+    Doc {
+        #[command(flatten)]
+        file: FileArgs<DocOutputFormat>,
+        /// The page order when documents with layouts that expand in horizontal and vertical directions are cut into pages.
+        #[arg(short = 'P', long, default_value_t = PageOrder::default())]
+        page_order: PageOrder,
+    },
+    /// Export each page of the documents individually.{n}
+    /// Both --output-dir and --output-format need to be set.{n}
+    #[command(alias = "pages")]
+    DocPages {
+        /// the directory the pages get exported to
+        #[arg(short = 'o', long)]
+        output_dir: PathBuf,
+        /// The directory the pages get exported to.{n}
+        /// Uses file stem of input file if not set.{n}
+        /// Cannot be used when exporting multiple files
+        #[arg(short = 's', long)]
+        output_file_stem: Option<String>,
+        /// the export output format
+        #[arg(short = 'f', long)]
+        output_format: DocPagesOutputFormat,
+        /// The page order when documents with layouts that expand in horizontal and vertical directions are cut into pages
+        #[arg(short = 'P', long, default_value_t = PageOrder::default())]
```

Comment:
Considering that this is not an option that is that frequently used, the short version can be removed. It avoid cluttering the namespace with `-p` *and* `-P`.